research/minigo/README.md

```diff
@@ -51,64 +57,82 @@ The value head applies the following modules:
+    │       │      └── full       # full sgf files of model selfplay
+    │       ├── ...
+    │       └── evaluate          # clean sgf files of model evaluation
+    │
+    └── ...
 
 ## Validating Model
- Run `minigo.py` with `--validation` argument
+To validate the trained model, issue the following command with `--validation` argument:
  ```
- python minigo.py --validation
- ```
- The `--validation` argument is to generate holdout dataset for model validation
-
-## Evaluating MiniGo Models
- Run `minigo.py` with `--evaluation` argument
+ python minigo.py --base_dir=$HOME/minigo/ --board_size=9 --batch_size=256 --validation
  ```
- python minigo.py --evaluation
- ```
- The `--evaluation` argument is to invoke the evaluation between the latest model and the current best model.
-
-## Testing Pipeline
-As the whole RL pipeline may takes hours to train even for a 9x9 board size, we provide a dummy model with a `--debug` mode for testing purpose.
 
- Run `minigo.py` with `--debug` argument
- ```
- python minigo.py --debug
- ```
- The `--debug` argument is for testing purpose with a dummy model.
+## Evaluating Models
+The performance of two models are compared with evaluation step. Given two models, one plays black and the other plays white. They play several games (# of games can be configured by parameter `eval_games` in [model_params.py](model_params.py)), and the one wins by a margin of 55% will be the winner.
 
-Validation and evaluation can also be tested with the dummy model by combing their corresponding arguments with `--debug`.
-To test validation, run the following commands:
- ```
- python minigo.py --debug --validation
- ```
-To test evaluation, run the following commands:
+To include the evaluation step in the RL pipeline, `--evaluation` argument can be specified to compare the performance of the `current_trained_model` and the `best_model_so_far`. The winner is used to update `best_model_so_far`. Run the following command to include evaluation step in the pipeline:
  ```
- python minigo.py --debug --evaluation
- ```
-To test both validation and evaluation, run the following commands:
- ```
- python minigo.py --debug --validation --evaluation
+ python minigo.py --base_dir=$HOME/minigo/ --board_size=9 --batch_size=256 --evaluation
  ```
 
-## MCTS and Go features (TODO)
-Code clean up on MCTS and Go features.
+## Testing Pipeline
+As the whole RL pipeline may take hours to train even for a 9x9 board size, a `--test` argument is provided to test the pipeline quickly with a dummy neural network model.
+
+To test the RL pipeline with a dummy model, issue the following command:
+```
+ python minigo.py --base_dir=$HOME/minigo/ --board_size=9 --batch_size=256 --test
+```
+
+## Running Self-play Only
+Self-play only option is provided to run selfplay step individually to generate training data in parallel. Issue the following command to run selfplay only with the latest trained model:
+```
+ python minigo.py --selfplay_only
```

Comment:
usually --selfplay is clear enough as a flag name. Word "only" is bit redundant.

Reply:
Sure. Will update it.